spec/support/custom_matchers/be_same_time_as.rb

```diff
@@ -1,5 +1,13 @@
 RSpec::Matchers.define :be_same_time_as do |expected|
+  regexp = /(.*_spec\.rb:\d+)/
+  called_from = caller.detect { |line| line =~ regexp }
+
   match do |actual|
+    puts <<-MESSAGE
+\rWARNING: The `be_same_time_as` matcher is deprecated and will be removed shortly.
+Use the `be_within` matcher instead: `be_same_time_as(expected_time).precision(1) == be_within(0.1).of(expected_time)`
```

Comment:
Incidentally, I don't think there were any callers that actually used the precision bit.  It is defaulted below as 5, meaning 5 digits of precision, so this example should be `be_within(0.00001).of(expected_time)`

Reply:
There are a few that use it, sadly, so thought I'd use the full functionality of the matcher using the precision.

I didn't do exactly the `0.0001` for a direct `be_same_time_as` (without `precision`) because anyone using it probably doesn't care and just wants seconds, so `0.1` is fine and what people would likely write anyway.